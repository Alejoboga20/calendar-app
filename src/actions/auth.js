import Swal from 'sweetalert2';
import { fetchWithoutToken, fetchWithToken } from '../helpers/fetch';
import { types } from '../types/types';

export const startLogin = (email, password) => {
  return async (dispatch) => {
    const response = await fetchWithoutToken(
      'auth',
      { email, password },
      'POST'
    );

    const body = await response.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      const { uid, name } = body;

      dispatch(login({ uid, name }));
    } else {
      Swal.fire('Error', body.msg, 'error');
    }
  };
};

export const startRegister = (name, email, password) => {
  return async (dispatch) => {
    const response = await fetchWithoutToken(
      'auth/new',
      { name, email, password },
      'POST'
    );

    const body = await response.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      const { uid, name } = body;

      dispatch(login({ uid, name }));
    } else {
      Swal.fire('Error', body.msg, 'error');
    }
  };
};

export const startChecking = () => {
  return async (dispatch) => {
    const response = await fetchWithToken('auth/renew');

    const body = await response.json();

    if (body.ok) {
      localStorage.setItem('token', body.token);
      localStorage.setItem('token-init-date', new Date().getTime());

      const { uid, name } = body;

      dispatch(login({ uid, name }));
    } else {
      dispatch(checkingFinish());
    }
  };
};

const checkingFinish = () => ({ type: types.authCheckingFinish });

const login = (user) => ({
  type: types.authLogin,
  payload: user
});
