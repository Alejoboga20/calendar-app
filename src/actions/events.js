import Swal from 'sweetalert2';
import { fetchWithToken } from '../helpers/fetch';
import { prepareEvents } from '../helpers/prepareEvents';
import { types } from '../types/types';

export const eventStartAddNew = (event) => {
  return async (dispatch, getState) => {
    const { uid, name } = getState().auth;

    try {
      const response = await fetchWithToken('event', event, 'POST');
      const body = await response.json();

      if (body.ok) {
        event.id = body.event.id;
        event.user = {
          _id: uid,
          name: name
        };
        dispatch(eventAddNew(event));
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventAddNew = (event) => ({ type: types.eventAddNew, payload: event });

export const eventSetActive = (event) => ({
  type: types.eventSetActive,
  payload: event
});

export const eventClearActiveEvent = () => ({
  type: types.eventClearActiveEvent
});

export const eventStartUpdate = (event) => {
  return async (dispatch) => {
    const eventId = event.id;

    try {
      const response = await fetchWithToken(`event/${eventId}`, event, 'PUT');
      const body = await response.json();

      if (body.ok) {
        dispatch(eventUpdated(event));
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventUpdated = (event) => ({
  type: types.eventUpdated,
  payload: event
});

export const eventStartDelete = (event) => {
  return async (dispatch, getState) => {
    const eventId = getState().calendar.activeEvent.id;
    console.log(eventId);

    try {
      const response = await fetchWithToken(`event/${eventId}`, {}, 'DELETE');
      const body = await response.json();

      if (body.ok) {
        dispatch(eventDeleted());
      } else {
        Swal.fire('Error', body.msg, 'error');
      }
    } catch (error) {
      console.log(error);
    }
  };
};

const eventDeleted = () => ({ type: types.eventDeleted });

export const eventStartLoading = () => {
  return async (dispatch) => {
    try {
      const response = await fetchWithToken('event');
      const body = await response.json();

      const events = prepareEvents(body.events);

      dispatch(eventLoaded(events));
    } catch (error) {
      console.log(error);
    }
  };
};

const eventLoaded = (events) => ({ type: types.eventLoaded, payload: events });

export const eventLogout = () => ({ type: types.eventLogout });
