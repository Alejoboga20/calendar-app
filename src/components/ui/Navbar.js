import React from 'react';

export const Navbar = () => {
  return (
    <div className='navbar navbar-dark bg-dark mb-4'>
      <span className='navar-brand'>Alejo</span>
      <button className='btn btn-danger'>
        <span className='fas fa-sign-out-alt'> Exit</span>
      </button>
    </div>
  );
};
