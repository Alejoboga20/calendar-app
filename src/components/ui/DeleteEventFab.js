import React from 'react';
import { useDispatch } from 'react-redux';
import { eventStartDelete } from '../../actions/events';

export const DeleteEventFab = () => {
  const dispatch = useDispatch();

  const handleDelete = () => {
    dispatch(eventStartDelete());
  };

  return (
    <button onClick={handleDelete} className='btn btn-danger fab-danger'>
      <i className='fas fa-trash'></i>
      <span>Delete Event</span>
    </button>
  );
};
