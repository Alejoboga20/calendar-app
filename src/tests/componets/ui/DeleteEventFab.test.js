import React from 'react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { DeleteEventFab } from '../../../components/ui/DeleteEventFab';
import { eventStartDelete } from '../../../actions/events';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = {};
const store = mockStore(initState);
store.dispatch = jest.fn();

jest.mock('../../../actions/events', () => ({
  eventStartDelete: jest.fn()
}));

const wrapper = mount(
  <Provider store={store}>
    <DeleteEventFab />
  </Provider>
);

describe('DeleteEventFab Tests', () => {
  test('should display properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call eventStartDelete', () => {
    wrapper.find('button').prop('onClick')();
    expect(eventStartDelete).toHaveBeenCalled();
  });
});
