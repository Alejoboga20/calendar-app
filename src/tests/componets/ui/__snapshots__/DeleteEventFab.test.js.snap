// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteEventFab Tests should display properly 1`] = `
<button
  className="btn btn-danger fab-danger"
  onClick={[Function]}
>
  <i
    className="fas fa-trash"
  />
  <span>
    Delete Event
  </span>
</button>
`;
